plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -434,6 +428,19 @@ public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggrega
         return Optional.of(new AggregationApplicationResult<>(handle, projections.build(), resultAssignments.build(), ImmutableMap.of(), precalculateStatisticsForPushdown));
     }
 
+    @VisibleForTesting
```

Comment:
see https://github.com/trinodb/trino/pull/18984#discussion_r1326801269

for base-jdbc we've settled on inheritance since it makes implementing new connectors based on it super duper easy.

Reply:
also if i remove the "unit test" and convert to an integration test there won't be this `VisibleForTesting` needed.

Since the integration test that I came up with (a query with too many aggregation functions so that new aliases get generated) is very "wordy" and also prone to breaking due to planner changes I opted for the unit test + VisibleForTesting.